docker/secondary-analysis-python/tools/submit/submit/stage.py

```diff
@@ -0,0 +1,134 @@
+        parser.add_argument('file_path', metavar="<file>",
+                            help="name of file to stage")
+        parser.add_argument('urn', metavar='<URN>',
+                            help="URN of staging area (given to you by Ingest Broker)")
+        self.args = parser.parse_args()
+
+    def _parse_urn(self, urn):
+        # TODO reimplement as deque
+        urnbits = urn.split(':')
+        if len(urnbits) == 6:
+            self.deployment_stage = urnbits[3]
+            self.area_uuid = urnbits[4]
+            encoded_credentials = urnbits[5]
+        elif len(urnbits) == 5:
+            self.deployment_stage = 'dev'
+            self.area_uuid = urnbits[3]
+            encoded_credentials = urnbits[4]
+        else:
+            raise RuntimeError("Bad URN")
+        uppercase_credentials = json.loads(base64.b64decode(encoded_credentials))
+        self.aws_credentials = {k.lower(): v for k, v in uppercase_credentials.items()}
+
+    def callback(self, bytes_transferred):
+        self.cumulative_bytes_transferred += bytes_transferred
+        percent_complete = (self.cumulative_bytes_transferred * 100) / self.file_size
+        sys.stdout.write("\r%s of %s transferred (%.0f%%)%s" %
+                         (sizeof_fmt(self.cumulative_bytes_transferred),
+                          sizeof_fmt(self.file_size),
+                          percent_complete,
+                          self.CLEAR_TO_EOL))
+        sys.stdout.flush()
+
+    def _stage_file(self, file_path):
+        print("Uploading %s to staging area %s..." % (os.path.basename(file_path), self.area_uuid))
+        self.file_size = os.stat(file_path).st_size
+        bucket_name = self.STAGING_BUCKET_TEMPLATE % (self.deployment_stage,)
+        file_s3_key = "%s/%s" % (self.area_uuid, os.path.basename(file_path))
+        bucket = self.s3.Bucket(bucket_name)
+        obj = bucket.Object(file_s3_key)
+        with open(file_path, 'rb') as fh:
+            self.cumulative_bytes_transferred = 0
+            obj.upload_fileobj(fh,
+                               ExtraArgs={'ACL': 'bucket-owner-full-control'},
+                               Callback=self.callback,
+                               Config=self.transfer_config(self.file_size)
+                               )
+        print("\n")
+
+    @classmethod
+    def transfer_config(cls, file_size):
+        etag_stride = cls._s3_chunk_size(file_size)
+        return TransferConfig(multipart_threshold=etag_stride,
+                              multipart_chunksize=etag_stride)
+
+    @staticmethod
+    def _s3_chunk_size(file_size):
+        if file_size <= 10000 * 64 * MB:
+            return 64 * MB
+        else:
+            div = file_size // 10000
+            if div * 10000 < file_size:
+                div += 1
+            return ((div + (MB-1)) // MB) * MB
+
+def run():
+    runner = Main()
+
+if __name__ == '__main__':
+    runner = Main()
```

Comment:
convention is to use `_` or `unused` as variable names for classes whose return values are not used. 

Alternatively, you could just call `Main()` and not assign it.